internal/gensupport/send.go

```diff
@@ -43,6 +44,12 @@ func (e wrappedCallErr) Is(target error) bool {
 // req.WithContext, then calls any functions returned by the hooks in
 // reverse order.
 func SendRequest(ctx context.Context, client *http.Client, req *http.Request) (*http.Response, error) {
+	// Add headers set in context metadata.
+	headers := callctx.HeadersFromContext(ctx)
+	for k, v := range headers {
+		req.Header[k] = append(req.Header[k], v...)
```

Comment:
Maybe use the [`Header.Add`](https://pkg.go.dev/net/http#Header.Add) helper instead? Does the same thing, but case insensitive/uses Canonical header helper.

Reply:
Ah good point, I was trying to avoid having to make a nested loop to iterate through keys but I didn't think of the case thing. Will change this.